### PR TITLE
Enforce minimum value of zero on max history size

### LINKED ISSUE #217: Max history size option can be negative
## Problem
The "Max history size" option can go negative, which doesn't make sense
image

The value should either be a positive integer, or zero (essentially disabling history).

### PR DESCRIPTION
**Pull request by ChiefOfGxBxL**
## Maximum history size minimum value bug
Resolves an issue where the user can specify a maximum history size that goes negative, e.g. `historySize = -5`.

This is problematic not just for aesthetics, but because the fetching-history logic uses `.slice(0, historySize)`, so a negative history size will yield results.

Closes #217

### CODE REVIEW

File: src/components/OptionsModal/GeneralOptionsPane.js [hunk 1]
```diff
@@ -79,9 +79,17 @@ function GeneralOptionsPane({ options, updateOption }) {
...
                     onChange={e => {
-                      updateOption('historySize', e.target.value);
+                      try {
+                        let value = parseInt(e.target.value, 10);
+                        if (value < 0) value = 0; // prevents manually typing negative value
+                        updateOption('historySize', value);
+                      } catch (parseIntException) {
+                        // Ignore any attempts to type an invalid value,
+                        // like letters, or a negative sign
+                      }
```
Comment (esphen) [inline src/components/OptionsModal/GeneralOptionsPane.js:92]: This is great! Let me suggest simplifying it slightly though. We can use `e.target.valueAsNumber` as `FormControl` is a number input (`type=number`) - so it exposes that special property which returns a number directly.

In theory a number input should not dispatch any update events unless the input is a valid number, but in practice you may get `NaN` back sometimes like when you try to clear the input so it never hurts to be safe. `isNaN` should guard against anything that `valueAsNumber` can return that we don't want.

```suggestion
                      const value = e.target.valueAsNumber;
                      // Prevent invalid input
                      if (isNaN(value)) return;
                      updateOption('historySize', value >= 0 ? value : 0);
```

**Review by esphen — COMMENTED** [review]
Looks great! Have a look at my suggestion below before we merge this.

File: src/components/OptionsModal/GeneralOptionsPane.js (hunk 1 above)
Comment (ChiefOfGxBxL) [reply src/components/OptionsModal/GeneralOptionsPane.js:92]: Cool, I didn't know `e.target.valueAsNumber` existed. Is that native JS, or is React adding it as a handy feature? Makes this code much simpler.

**Comment by esphen** [comment]
Once again thanks for the great contribution!

**Pull request merged**